Secret-agency-web/src/main/react/secret-agency/src/components/departments-page/DepartmentsPage.tsx

```diff
@@ -62,10 +64,42 @@ export class DepartmentsPage extends React.Component<{}, IDepartmentsState> {
             longitude: parseFloat(longitude),
             specialization
         };
-        const createdDepartment = await createDepartment(department);
-        this.setState((prevState) => ({
-            departments: prevState.departments.set(createdDepartment.id, createdDepartment)
-        }));
+        const errors = this.validate(department);
+        if (errors.length === 0) {
+            const createdDepartment = await createDepartment(department);
+            this.setState((prevState) => ({
+                departments: prevState.departments.set(createdDepartment.id, createdDepartment)
+            }));
+            this.setState(_ => ({editedDepartmentId: null}));
```

Comment:
Tohle ti zmerguje hodnoty? Nepřepíše to ty původni?

Reply:
nasetuje to jen `editedDepartmentId`